django/contrib/admin/options.py

```diff
@@ -1522,7 +1522,7 @@ def changelist_view(self, request, extra_context=None):
         if (request.method == "POST" and cl.list_editable and
                 '_save' in request.POST and not action_failed):
             FormSet = self.get_changelist_formset(request)
-            formset = cl.formset = FormSet(request.POST, request.FILES, queryset=cl.result_list)
+            formset = cl.formset = FormSet(request.POST, request.FILES, queryset=self.model._default_manager.get_queryset())
```

Comment:
Does this introduce an issue where a user can edit any object in the queryset (outside of filtering done by `ModelAdmin.get_queryset()`, for example?